lms/static/js/bookmarks/views/bookmarks_list.js

```diff
@@ -0,0 +1,133 @@
+            loadingIcon: '<i class="fa fa-fw fa-spinner fa-pulse message-in-progress" aria-hidden="true"></i>',
+
+            errorMessage: gettext('An error has occurred. Please try again.'),
+            loadingMessage: gettext('Loading'),
+
+            url: '/api/bookmarks/v0/bookmarks/',
+
+            events : {
+                'click .bookmarks-results-list-item': 'visitBookmark'
+            },
+
+            initialize: function (options) {
+                this.template = _.template($('#bookmarks_list-tpl').text());
+                this.loadingMessageView = options.loadingMessageView;
+                this.errorMessageView = options.errorMessageView;
+                this.courseId = $(this.el).data('courseId');
+                this.langCode = $(this.el).data('langCode');
+                _.bindAll(this, 'render', 'breadcrumbTrail', 'humanFriendlyDate', 'createBookmarkUrl');
+            },
+
+            render: function () {
+                var data = {
+                    bookmarks: this.collection.models,
+                    breadcrumbTrail: this.breadcrumbTrail,
+                    humanFriendlyDate: this.humanFriendlyDate,
+                    createBookmarkUrl: this.createBookmarkUrl
+                };
+                this.$el.html(this.template(data));
+                this.delegateEvents();
+                return this;
+            },
+
+            showBookmarksList: function () {
+                var view = this;
+
+                this.hideErrorMessage();
+                this.showBookmarksContainer();
+                this.showLoadingMessage();
+
+                this.collection.url = this.url;
+                this.collection.fetch({
+                    reset: true,
+                    data: {course_id: this.courseId, fields: 'display_name,path'}
+                }).done(function () {
+                    view.hideLoadingMessage();
+                    view.render();
+                    view.focusBookmarksElement();
+                }).fail(function () {
+                    view.hideLoadingMessage();
+                    view.showErrorMessage();
+                });
+            },
+
+            visitBookmark: function (event) {
+                window.location = event.target.pathname;
+            },
+
+            /**
+             * Create a breadcrumb trail from section, subsection and unit display names.
+             * @param {Array} bookmarkPath - Contains 2 objects of info related to section and subsection.
+             * @param {String} unitDisplayName - Unit display name.
+             */
+            breadcrumbTrail: function (bookmarkPath, unitDisplayName) {
+                var separator = ' <i class="icon fa fa-caret-right" aria-hidden="true"></i><span class="sr">-</span> ';
+                var names = _.pluck(bookmarkPath, 'display_name');
+                names.push(unitDisplayName);
+                return names.join(separator);
+            },
```

Comment:
Since this logic has to do with html generation, can we not move it to the template? It should be possible to do it in a single line.